models/actions/book/user-goals/UpdateBuyer.model.bxb

```diff
@@ -0,0 +1,74 @@
+action (UpdateBuyer) {
+  description (Prompt user to enter personal data)
+  type (Constructor)
+  collect {
+    //Order is a structure with transaction feature that has the current buyer and product properties
+    input (order) {
+      type (Order)
+      min (Required)
+    }
+    //This is the list of the buyer fields which user wants to update
+    input (fieldsToUpdate) {
+      type (self.Field)
+      min (Required) max (Many)
+      default-init {
+        intent {
+          goal-set: self.Field {
+            self.Field(firstName)
+            self.Field(lastName)
+            self.Field(emailInfo)
+            self.Field(phoneInfo)
+          }
+         }
+      }
+    }
+    //new value for first name
+    input (firstName) {
+      type (contact.FirstName)
+      min (Optional) max (One)
+    }
+    //new value for last name
+    input (lastName) {
+      type (contact.LastName)
+      min (Optional) max (One)
+    }
+    //new value for email address
+    input (emailAddress) {
+      type (contact.EmailAddress)
+      min (Optional) max (One)
+    }
+    //new value for phone number
+    input (phoneNumber) {
+      type (contact.PhoneNumber)
+      min (Optional) max (One)
+    }
+    input (forcePrompt) {
+      type (self.ForcePrompt)
+      min (Optional) max (One)
+    }
+    //compute the new buyer information by calling self library
+    computed-input (buyer) {
+      type (Buyer)
+      min (Required)
+      compute {
+        intent {
+          goal: self.GetProfile
+          value: self.Profile$expr(order.buyer)
+          //do not force a prompt if a field value is available. make sure to include all fields.
```

Comment:
Is this comment still relevant? I don't see the conditional check that used to go with it